src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -4553,13 +4633,33 @@ private void onFeatureExtAck(byte featureId, byte[] data) throws SQLServerExcept
                     throw new SQLServerException(SQLServerException.getErrString("R_InvalidAEVersionNumber"), null);
                 }
 
-                byte supportedTceVersion = data[0];
-                if (0 == supportedTceVersion || supportedTceVersion > TDS.MAX_SUPPORTED_TCE_VERSION) {
+                aeVersion = data[0];
+                if (0 == aeVersion || aeVersion > TDS.COLUMNENCRYPTION_VERSION2) {
                     throw new SQLServerException(SQLServerException.getErrString("R_InvalidAEVersionNumber"), null);
                 }
 
-                serverSupportsColumnEncryption = true;
+                serverColumnEncryptionVersion = ColumnEncryptionVersion.AE_v1;
+
+                if (null != enclaveAttestationUrl) {
+                    if (aeVersion < TDS.COLUMNENCRYPTION_VERSION2) {
+                        throw new SQLServerException(SQLServerException.getErrString("R_enclaveNotSupported"), null);
+                    } else {
+                        serverColumnEncryptionVersion = ColumnEncryptionVersion.AE_v2;
+                        enclaveType = new String(data, 2, data.length - 2, UTF_16LE);
+                    }
+
+                    if (null == enclaveType) {
```

Comment:
I don't think this check makes sense, `new String()` will set  `enclaveType` to a non-null value. Better perform the check on `data`.

Reply:
data is not null